secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/common/rest/ContentTypeNegotiator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.common.rest;
+
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.forgerock.http.util.MultiValueMap;
+import org.forgerock.util.Reject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Can be used to determine the most appropriate media type for the Entity in a response
+ */
+public class ContentTypeNegotiator {
+    private final Logger logger = LoggerFactory.getLogger(ContentTypeNegotiator.class);
+
+    private final TreeMap<Float, List<String>> mediaTypesMap;
+    
+    private final List<String> supportedContentTypes;
+
+    public ContentTypeNegotiator(List<String> supportedContentTypes) {
+        Reject.ifNull(supportedContentTypes, "supportedContentTypes must not be null");
+        if(supportedContentTypes.isEmpty()){
+            throw new IllegalArgumentException("supportedContentTypes must not be empty");
+        }
+        mediaTypesMap = new TreeMap<>();
```

Comment:
This doesn't look threadsafe.

The MultiValueMap are wrapping this map and modifying its contents per call to buildMapOfAcceptableMediaTypes.